packages/config/src/files.js

```diff
@@ -26,11 +27,17 @@ const resolvePaths = function (config, propNames, baseRel) {
 
 const resolvePathProp = function (config, propName, baseRel) {
   const path = get(config, propName)
-  return path === undefined ? config : set(config, propName, resolvePath(baseRel, path))
+
+  if (!isTruthy(path)) {
+    deleteProp(config, propName)
```

Comment:
This ensures `@netlify/config` return empty file paths as `undefined` instead of sometimes empty strings, sometimes `undefined`.